src/common/io.ts

```diff
@@ -1235,10 +1270,29 @@ export class ARIOWriteable extends ARIOReadable implements AoARIOWrite {
     });
   }
 
+  private async sendArNSPurchaseIntentToTurbo(
+    params: InitiateArNSPurchaseParams,
+    options?: WriteOptions,
+  ): Promise<AoMessageResult> {
+    const { arioWriteResult, purchaseReceipt } =
+      await this.fundFromTurbo.initiateArNSPurchase(params, options);
+    return { ...arioWriteResult, receipt: purchaseReceipt };
+  }
+
   async buyRecord(
     params: AoBuyRecordParams,
     options?: WriteOptions,
   ): Promise<AoMessageResult> {
+    if (params.fundFrom === 'turbo') {
+      return this.sendArNSPurchaseIntentToTurbo({
+        intent: 'Buy-Record',
```

Comment:
Buy-Name is the new action, Buy-Record is deprecated
```suggestion
        intent: 'Buy-Name',
```

Reply:
[c142951](https://github.com/ar-io/ar-io-sdk/pull/450/commits/c142951c6458a55fde7f10c90fa92c77544b7600)